specification/elastic/resource-manager/Microsoft.Elastic/preview/2022-09-01-preview/elastic.json

```diff
@@ -0,0 +1,2308 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "AllTrafficFilters_list": {
+            "$ref": "./examples/AllTrafficFilters_list.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Elastic/monitors/{monitorName}/listAssociatedTrafficFilters": {
+      "post": {
+        "tags": [
+          "TrafficFilter"
+        ],
+        "summary": "Get the list of all associated traffic filters for the given deployment.",
+        "operationId": "listAssociatedTrafficFilters_list",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/MonitorNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ElasticTrafficFilterResponse"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ResourceProviderDefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "listAssociatedTrafficFilters_list": {
+            "$ref": "./examples/AssociatedFiltersForDeployment_list.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Elastic/monitors/{monitorName}/createAndAssociateIPFilter": {
+      "post": {
+        "tags": [
+          "TrafficFilter"
+        ],
+        "summary": "Create and Associate IP traffic filter for the given deployment.",
+        "operationId": "createAndAssociateIPFilter_Create",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/MonitorNameParameter"
+          },
+          {
+            "in": "query",
+            "name": "ips",
+            "description": "List of ips",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "name",
+            "description": "Name of the traffic filter",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Success"
+          },
+          "202": {
+            "description": "Success"
```

Comment:
>Success [](http://example.com/codeflow?start=28&length=7)

pl describe the schema of the final response of the 202 using a 200 status code. Also describe the final state using x-ms-long-running-operation-options.

Reply:
Do this everywhere

Reply:
@rkmanda do you mean we should add 200 status code as well along with 201 & 202??

Also, regarding describing the final state using x-ms-long-running-operation-options, this functionality of creating a IP traffic filter is not new, this was introduced in API version 2022-07-01-preview. Could you please let us know if this is really required because PR for previous version when this was introduced was accepted without this.

Reply:
@rkmanda could you please help us here?

Reply:
For your final response, what will be returned? Is there no response? And that's why it's returning 201?

Reply:
@zizw123 We return 201 and we don't have any response body. As per the CLI docs 202 response is necessary for all x-ms-long-running-operation: true (as mentioned in [link](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#202-accepted-and-location-headers) ) which was missed in the previous version. Hence we added it in both the versions 2022-07-01-preview & 2022-09-01-preview.

Error message received in CLI
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/106945698/207592481-e7e7f905-cf9c-4dbd-bc9c-2911b8d55dbe.png)

Note: this error is not received when the call is made via Azure Portal or Postman.
Please let me know if you need more info on this, we can have a quick call.

Reply:
@pachaturvedi As rkmanda mentioned, you should use 200 and 202, see https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously

Reply:
@zizw123 update the status code in new and previous version. We'll make the respective changes in our RP code as well.